internal/pkg/cli/task_run_test.go

```diff
@@ -455,29 +461,56 @@ func TestTaskRunOpts_Execute(t *testing.T) {
 		"error updating resources": {
 			setupMocks: func(ctrl *gomock.Controller) {
 				mockDeployer = mocks.NewMocktaskDeployer(ctrl)
+				mockRepo = mocks.NewMockrepositoryService(ctrl)
 
 				mockDeployer.EXPECT().DeployTask(&deploy.CreateTaskResourcesInput{
-					Name: inGroupName,
+					Name:  inGroupName,
 					Image: "",
 				}).Return(nil)
+				mockRepo.EXPECT().BuildAndPush(gomock.Any(), gomock.Any(), imageTagLatest)
+				mockRepo.EXPECT().URI().Return(mockRepoURI)
 				mockDeployer.EXPECT().DeployTask(&deploy.CreateTaskResourcesInput{
 					Name: inGroupName,
-					// TODO: use image.URI() from mockRepository
+					Image: fmt.Sprintf(fmtImageURI, mockRepoURI, imageTagLatest),
 				}).Times(1).Return(errors.New("error updating"))
 			},
 			wantedError: fmt.Errorf("update resources for task %s: %w", inGroupName, errors.New("error updating")),
```

Comment:
i think it's better to write the expected error explicitly like `update resources for task my-task: error updating` because unit tests are meant to prevent us from doing crazy things and we should include as little logic in expected ones as possible i think. Just a nit tho